core/src/layers/complete.rs

```diff
@@ -380,13 +380,14 @@ impl<A: Access> LayeredAccess for CompleteAccessor<A> {
         &self.inner
     }
 
-    fn metadata(&self) -> AccessorInfo {
-        let mut meta = self.meta.clone();
+    // Todo: May move the logic to the implement of Layer::layer of CompleteAccessor<A>
```

Comment:
Nice idea, we can create an issue for this.

Reply:
I think the issue can be create after this refactor is done, because this may related to the https://github.com/apache/opendal/issues/4845#issuecomment-2224641607